photon-client/src/main/scala/com/linkedin/photon/ml/estimators/GameEstimatorEvaluationFunction.scala

```diff
@@ -33,21 +36,44 @@ import com.linkedin.photon.ml.util.DoubleRange
  * @param baseConfig The initial configuration supplied by the user
  * @param data Training data
  * @param validationData Validation data
- * @param ranges Hyper-parameter searching range. If search space is [a, b], range here is [log(a), log(b)].
  */
 class GameEstimatorEvaluationFunction(
     estimator: GameEstimator,
     baseConfig: GameOptimizationConfiguration,
     data: DataFrame,
-    validationData: DataFrame,
-    ranges: Seq[DoubleRange])
+    validationData: DataFrame)
   extends EvaluationFunction[GameResult] {
 
+  import GameEstimatorEvaluationFunction._
+
   // CoordinateOptimizationConfigurations sorted in order by coordinate ID name
   private val baseConfigSeq = baseConfig.toSeq.sortBy(_._1)
 
+  // Pull the hyperparameter ranges from the optimization configuration
+  protected[estimators] val ranges = baseConfigSeq
+    .flatMap {
+      case (_, config: GLMOptimizationConfiguration) => {
+        val regularizationWeightRange = config.regularizationWeightRange
+          .getOrElse(DEFAULT_REG_WEIGHT_RANGE)
+          .transform(log)
+
+        val elasticNetParamRange = config.elasticNetParamRange
+          .getOrElse(DEFAULT_REG_ALPHA_RANGE)
```

Comment:
Fits on one line

Reply:
I fixed all the other ones, but I don't want to fix this one. If I put this all on one line, it doesn't match the initialization of `regularizationWeightRange` on L55 stylistically and looks super weird and IMO harder to read.

Reply:
10,000 years gulag